### DISCUSSION TITLE
[Flutter] [Sign in with Google] Multiple fingerprints & client ids to appAuthRedirectScheme

### DISCUSSION THREAD

**Discussion by romaingyh** [supabase/supabase#16060] [Questions]
Hello, I'm trying to implement Sign in with Google in my Flutter app but I've an issue with android part.

Doc say to create android Client ID from app sha1 fingerprint and then add the reverse DNS form of the client id to `appAuthRedirectScheme` in `build.gradle`.

But in Android we have two different fingerprints : debug and release, and even 3 fingerprints with the google play fingerprint when app is published.
So which one to choose ? Should I have to create 3 differents client ids in google cloud ? But if so, what should I put in `appAuthRedirectScheme` ?

Thanks !

**Comment by dshukertjr** [comment]
When going to production on Android, you may have multiple SHA-1 fingerprint certificates, debug and the production certificate provided from Play Store. You will need to create one Google OAuth 2.0 Client ID for each certificates. You can then utilize [flavors](https://docs.flutter.dev/deployment/flavors#using-flavors-in-android) to dynamically update the `appAuthRedirectScheme` like this:

```groovy
flavorDimensions "default"
productFlavors {
    dev {
        dimension "default"
        versionNameSuffix ".dev"
        manifestPlaceholders += [
            'appAuthRedirectScheme': 'com.googleusercontent.apps.*DEBUG_CLIENT_ID*'
        ]
    }
    prod {
        dimension "default"
        manifestPlaceholders += [
            'appAuthRedirectScheme': 'com.googleusercontent.apps.*PROD_CLIENT_ID*'
        ]
    }
}
```

Using flavors is probably not the only way to support multiple client IDs, so if you can find other ways that you are more comfortable with, that would be great too!
